crates/swc_ecma_minifier/src/analyzer/storage/normal.rs

```diff
@@ -98,8 +101,10 @@ impl Storage for ProgramData {
 
                     match kind {
                         ScopeKind::Fn => {
-                            e.get_mut().is_fn_local = false;
-                            e.get_mut().used_by_nested_fn = true;
+                            if !var_info.used_recursively {
```

Comment:
I'm not sure if this if statement is correct

Reply:
The general idea is for following code:
```js
function a() {
  x(a)
}
module.export = a
```
`a` needs to be `fn_local` to be inlined. But I'm also not sure if this is right and I believe we should be extremely cautious for this.